float32[] actions
---
float32[] states
float32[] actions
float32[] time
bool success
string reason